### PR TITLE
Add message to the build log when a build is running as SYSTEM

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-22949](https://issues.jenkins-ci.org/browse/JENKINS-22949).

Until the linked issue was addressed in https://jenkins.io/changelog/#v2.62, the concept of build item authenticators was a lot more discoverable. So let's restore that part.

Unsure how best to phrase this. Tempted to add a link to a jenkins.io page with build authentication docs, but that may be a bit much for every build. Thoughts?

### Proposed changelog entries

* Add a log message to build logs when builds run with the virtual SYSTEM authentication.

### Submitter checklist

- [sort of] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Review by jvz — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
❤️

**Comment by oleg-nenashev** [comment]
Maybe we should add a killswitch to let admins to disable the behavior if they are sure about that. Otherwise they will be getting issues from their users about the instance configuration.

**Comment by daniel-beck** [comment]
> Maybe we should add a killswitch to let admins to disable the behavior if they are sure about that.

Probably more a phrasing thing; there should be nothing secret about what internal authentication the build is running as.

**Comment by jglick** [comment]
> that may be a bit much for every build

I would make this an administrative monitor. JENKINS-24513 actually proposed something more sophisticated.

**Comment by oleg-nenashev** [comment]
Administrative monitor is fine with me. The implementation should not require much time, and it will help admins. Common users cannot do much about this issue anyway

File: core/src/main/resources/hudson/model/Messages.properties [hunk 1]
```diff
@@ -221,6 +221,7 @@ Run.BuildAborted=Build was aborted
 Run.MarkedExplicitly=This record is explicitly marked to be kept.
 Run.Permissions.Title=Run
 Run.running_as_=Running as {0}
+Run.running_as_SYSTEM=This build is running using the Jenkins-internal SYSTEM authentication with no permission restrictions.
```
Comment (Wadeck) [inline core/src/main/resources/hudson/model/Messages.properties]: I can only +1 the idea to link a page / blog post about the recommended way to configure the AuthQueue.

**Review by Wadeck — APPROVED** [review]
:bee:

File: core/src/main/resources/hudson/model/Messages.properties (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/resources/hudson/model/Messages.properties]: Right, but probably not in every build log -- technically adding a `HyperlinkNote` is trivial, but it seems very noisy.

**Comment by daniel-beck** [comment]
Un-onholding this after the message makes no claims that may not be accurate anymore.

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**